File: crm_backend/models/call_metrics.js
const db = require('../db');

exports.create = async (data) => {
    const [result] = await db.query(
        `INSERT INTO call_metrics (
            user_id, total_calls, incoming_calls, outgoing_calls, missed_calls, 
            connected_calls, attempted_calls, total_duration_seconds,
            stage_fresh, stage_interested, stage_committed, stage_not_interested
        ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`,
        [
            data.user_id,
            data.total_calls || 0,
            data.incoming_calls || 0,
            data.outgoing_calls || 0,
            data.missed_calls || 0,
            data.connected_calls || 0,
            data.attempted_calls || 0,
            data.total_duration_seconds || 0,
            data.stage_fresh || 0,
            data.stage_interested || 0,
            data.stage_committed || 0,
            data.stage_not_interested || 0
        ]
    );
    return result.insertId;
};

exports.findAll = async () => {
    const [rows] = await db.query('SELECT * FROM call_metrics');
    return rows;
};

exports.findById = async (id) => {
    const [rows] = await db.query('SELECT * FROM call_metrics WHERE id = ?', [id]);
    return rows[0] || null;
};

exports.findByUserId = async (user_id) => {
    const [rows] = await db.query('SELECT * FROM call_metrics WHERE user_id = ?', [user_id]);
    return rows;
};

exports.update = async (id, data) => {
    const fields = [];
    const values = [];
    const columns = [
        'total_calls', 'incoming_calls', 'outgoing_calls', 'missed_calls', 
        'connected_calls', 'attempted_calls', 'total_duration_seconds',
        'stage_fresh', 'stage_interested', 'stage_committed', 'stage_not_interested'
    ];
    columns.forEach(col => {
        if (data[col] !== undefined) {
            fields.push(`${col} = ?`);
            values.push(data[col]);
        }
    });
    if (fields.length === 0) return;
    values.push(id);
    await db.query(`UPDATE call_metrics SET ${fields.join(', ')} WHERE id = ?`, values);
};

exports.delete = async (id) => {
    await db.query('DELETE FROM call_metrics WHERE id = ?', [id]);
}; 

File: crm_final/lib/models/call_metrics.dart
class CallMetrics {
  final int? id;
  final int userId;
  final int totalCalls;
  final int incomingCalls;
  final int outgoingCalls;
  final int missedCalls;
  final int connectedCalls;
  final int attemptedCalls;
  final int totalDurationSeconds;
  final int stageFresh;
  final int stageInterested;
  final int stageCommitted;
  final int stageNotInterested;

  CallMetrics({
    this.id,
    required this.userId,
    required this.totalCalls,
    required this.incomingCalls,
    required this.outgoingCalls,
    required this.missedCalls,
    required this.connectedCalls,
    required this.attemptedCalls,
    required this.totalDurationSeconds,
    required this.stageFresh,
    required this.stageInterested,
    required this.stageCommitted,
    required this.stageNotInterested,
  });

  factory CallMetrics.fromJson(Map<String, dynamic> json) {
    return CallMetrics(
      id: json['id'],
      userId: json['user_id'],
      totalCalls: json['total_calls'] ?? 0,
      incomingCalls: json['incoming_calls'] ?? 0,
      outgoingCalls: json['outgoing_calls'] ?? 0,
      missedCalls: json['missed_calls'] ?? 0,
      connectedCalls: json['connected_calls'] ?? 0,
      attemptedCalls: json['attempted_calls'] ?? 0,
      totalDurationSeconds: json['total_duration_seconds'] ?? 0,
      stageFresh: json['stage_fresh'] ?? 0,
      stageInterested: json['stage_interested'] ?? 0,
      stageCommitted: json['stage_committed'] ?? 0,
      stageNotInterested: json['stage_not_interested'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'total_calls': totalCalls,
      'incoming_calls': incomingCalls,
      'outgoing_calls': outgoingCalls,
      'missed_calls': missedCalls,
      'connected_calls': connectedCalls,
      'attempted_calls': attemptedCalls,
      'total_duration_seconds': totalDurationSeconds,
      'stage_fresh': stageFresh,
      'stage_interested': stageInterested,
      'stage_committed': stageCommitted,
      'stage_not_interested': stageNotInterested,
    };
  }
} 